app/WebParsing/UtsgJsonParser.hs

```diff
@@ -0,0 +1,148 @@
+{-# LANGUAGE OverloadedStrings, ScopedTypeVariables #-}
+
+
+module WebParsing.UtsgJsonParser
+     (insertAllCourses) where
+
+import Data.Aeson ((.:), (.:?), decode, FromJSON(..), Value(..))
+import Data.List
+import Database.Tables
+import WebParsing.PrerequisiteParsing
+import Data.Maybe
+import Data.String
+import Control.Monad
+import Database.Persist.Sqlite (runSqlite, insertMany_, insert_)
+import qualified Data.HashMap.Lazy as M
+import qualified Data.Text as T
+import Config (databasePath)
+import Network.HTTP.Conduit (simpleHttp)
+import Control.Applicative ((<$>), (<*>))
+
+import Data.Traversable
+import Data.Aeson.Types
+import qualified Data.HashMap.Strict as HM
+
+ -- | URL to UofT courses (stored as JSON string)
+jsonURL :: String
+jsonURL = "https://timetable.iit.artsci.utoronto.ca/api/courses?code=csc"
+
+-- | Converts 24-hour time into a double
+-- | Assumes times are rounded to the nearest hour
+getHourVal :: String -> Double
+getHourVal time = (read $ take 2 time :: Double) + (/) (read $ drop 3 time :: Double) 60
+
+-- | Converts a weekday into a double
+-- | Monday to Friday becomes 0.0 to 4.0
+getDayVal :: String -> Double
+getDayVal day = case day of
+                    "MO" -> 0.0
+                    "TU" -> 1.0
+                    "WE" -> 2.0
+                    "TH" -> 3.0
+                    "FR" -> 4.0
+
+-- | Takes a day and start/end times then generates a set of 30-minute timeslots
+getTimeSlots :: String -> String -> String -> [Time]
+getTimeSlots day start end = do
+    let dayDbl = getDayVal day
+        startDbl = getHourVal start
+        endDbl = getHourVal end
+    [Time [dayDbl, timeDbl] | timeDbl <- [startDbl, (startDbl + 0.5) .. (endDbl - 0.5)]]
+
+ -- | Decode JSON string into hash map object
+getJSON :: IO (Maybe DB)
+getJSON = do
+  resp <- simpleHttp jsonURL
+  return $ (decode resp :: Maybe DB)
+
+newtype Test = Test T.Text
+    deriving Show
+
+newtype Meeting = Meeting [(Either Lecture Test)] -- Tutorial
+  deriving Show
+
+lookupVal :: Value -> T.Text -> [T.Text]
+lookupVal (Object lst) val = map (\v -> case v of
+                                            (String s) -> s
+                                            _ -> "") $
+                                map (\(Object o) -> fromJust $ HM.lookup val o) $
+                                    map snd $ HM.toList lst
+
+instance FromJSON Meeting where
+    parseJSON = withObject "Meeting" $ \o -> do
+      code <- o .: "code"
+      session <- o .: "section"
+      meetings <- (o .: "meetings" :: Parser (HM.HashMap String (HM.HashMap String Value)))
+      return $ Meeting $ map (\(section, sectionHash) -> let (String cap) = fromJust $ HM.lookup "enrollmentCapacity" sectionHash
```

Comment:
@R0ss I looked into using Parser better. `sectionHash` is a `Value`. You can map a `withObject` onto the meeting sections, with a `Parser Lecture` in there.